### PR TITLE
Creates a Fun-O-Meter dashboard skeleton, showing breakdowns by day, script, and top and bottom levels.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Further work will allow seeing breakdowns by day for a specified script or script-level and will highlight scripts and script-levels with large changes in percentage.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
View of the dashboard when run locally (with only a small amount of local data).

image

File: dashboard/app/views/reports/funometer.html.haml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - @level_headers.each do |header|
+      %th
+        %span= header
+  - @hated_level_ratings.each do |level|
+    %tr
+      - level.each do |field|
+        %td
+          %span= field
+
+:javascript
```
Comment (Hamms) [inline dashboard/app/views/reports/funometer.html.haml]: Could you wrap this in a `- content_for :body_scripts do` block?

File: dashboard/app/controllers/reports_controller.rb [hunk 2]
```diff
@@ -56,6 +56,30 @@ def all_usage
...
+  def funometer
+    authorize! :read, :reports
+
+    # Compute the global funometer ratio.
+    all_ratings = PuzzleRating.all
+    positive_ratings = all_ratings.where(rating: 1)
+    @global_percentage = 100.0 * positive_ratings.count / all_ratings.count
+
+    # Construct the URL showing the daily trend.
+    percentages_by_day = all_ratings.select('100.0 * SUM(rating) / COUNT(rating)').where('created_at > ?', Time.now.prev_month).group('DATE(created_at)').order('DATE(created_at)').pluck('DATE(created_at)', '100.0 * SUM(rating) / COUNT(rating)')
```
Comment (Hamms) [inline dashboard/app/controllers/reports_controller.rb:68]: This appears to be the daily trend _for the last month_. Can that be specified somewhere?

File: dashboard/app/controllers/reports_controller.rb [hunk 3]
```diff
@@ -56,6 +56,30 @@ def all_usage
...
+
+    # Compute funometer ratios by script.
+    @script_headers = ['Script ID', 'Percentage', 'Count']
+    @script_ratings = all_ratings.select(:script_id, 'SUM(100.0 * rating) / COUNT(rating) AS ratio', 'COUNT(rating) AS cnt').group(:script_id).order('SUM(100.0 * rating) / COUNT(rating)').pluck(:script_id, 'SUM(100.0 * rating) / COUNT(rating)', 'COUNT(rating)')
```
Comment (Hamms) [inline dashboard/app/controllers/reports_controller.rb:72]: can you `.pluck(:script_id, 'ratio', 'cnt')` here, or does ActiveRecord not understand that?

File: dashboard/app/views/reports/funometer.html.haml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+%h1
+  Fun-O-Meter Stats
+
+Global Percentage:
+%span= number_to_percentage(@global_percentage, precision: 2)
+
+%br
+
+%div{id: "percentages_by_day"}
```
Comment (Hamms) [inline dashboard/app/views/reports/funometer.html.haml]: This line can just be `#percentages_by_day`

File: dashboard/app/views/reports/funometer.html.haml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+%script{src:'https://www.google.com/jsapi'}
+
+%h1
+  Fun-O-Meter Stats
+
+Global Percentage:
```
Comment (Hamms) [inline dashboard/app/views/reports/funometer.html.haml]: Is this text wrapped in a tag?

File: dashboard/app/views/reports/funometer.html.haml (hunk 1 above)
Comment (ashercodeorg) [inline dashboard/app/views/reports/funometer.html.haml]: Done.

File: dashboard/app/controllers/reports_controller.rb (hunk 2 above)
Comment (ashercodeorg) [inline dashboard/app/controllers/reports_controller.rb:68]: Improved the comment (as well as other comments [s/ratio/percentage]).

File: dashboard/app/views/reports/funometer.html.haml (hunk 4 above)
Comment (ashercodeorg) [inline dashboard/app/views/reports/funometer.html.haml]: Done.

File: dashboard/app/views/reports/funometer.html.haml (hunk 5 above)
Comment (ashercodeorg) [inline dashboard/app/views/reports/funometer.html.haml]: It is now.

**Comment by Hamms** [comment]
LGTM!

**Pull request merged**